lib/list_task.rb

```diff
@@ -23,10 +23,23 @@ def list(store, options = {})
       name = File.basename(name)
       begin
         description = SystemDescription.load(name, store, skip_validation: true)
+      rescue Machinery::Errors::SystemDescriptionIncompatible => e
+        if !e.format_version
+          Machinery::Ui.puts " #{name}: incompatible format version. Can not be upgraded.\n\n"
+        elsif e.format_version < SystemDescription::CURRENT_FORMAT_VERSION
+          Machinery::Ui.puts " #{name}: format version #{e.format_version}, " \
+            "needs to be upgraded.\n\n"
+        else
+          Machinery::Ui.puts " #{name}: format version #{e.format_version}. " \
+            "Please upgrade Machinery to the latest version.\n\n"
+        end
+        next
+      rescue Machinery::Errors::SystemDescriptionValidationFailed
+        Machinery::Ui.puts " #{name}: This description is broken. Use " \
+          "`#{$0} validate #{name}` to see the error message.\n\n"
+        next
       rescue Machinery::Errors::SystemDescriptionError
-        Machinery::Ui.puts " #{name}:\n"
```

Comment:
Machinery::Errors::SystemDescriptionError seems to be used quite often. Does this only happen if the json parser has issues?